onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -128,6 +129,45 @@ Status QnnBackendManager::LoadQnnSystemLib() {
   return Status::OK();
 }
 
+void QnnLogging(const char* format,
+                QnnLog_Level_t level,
+                uint64_t timestamp,
+                va_list argument_parameter) {
+  ORT_UNUSED_PARAMETER(level);
+  ORT_UNUSED_PARAMETER(timestamp);
+
+  std::ostringstream stream;
+  ::onnxruntime::logging::Capture::ProcessPrintfStream(stream, format, argument_parameter);
+  // Always output Qnn log as Ort verbose log
+  LOGS_DEFAULT(VERBOSE) << stream.str();
```

Comment:
nice, it is simpler now. I was thinking of manually creating a `Capture` instance and calling `Capture::ProcessPrintf()` directly. that could avoid the additional overhead of another stringstream. I think this should work though if the performance is fine.

Reply:
~Capture() will log the message. We don't want that.

Reply:
why not? that's what `LOGS_DEFAULT()` does, right?

Reply:
I get your point now.

---
In reply to: [1201128633](https://github.com/microsoft/onnxruntime/pull/16019#discussion_r1201128633) [](ancestors = 1201128633)

Reply:
done

---
In reply to: [1201194801](https://github.com/microsoft/onnxruntime/pull/16019#discussion_r1201194801) [](ancestors = 1201194801,1201128633)